forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabHost.java

```diff
@@ -210,4 +216,44 @@ public boolean isMemberOf(String groupId, HostUser user) {
                              .execute();
         return !details.isNull();
     }
+
+    CommitMetadata toCommitMetadata(JSONValue o) {
+        var hash = new Hash(o.get("id").asString());
+        var parents = o.get("parent_ids").stream()
+                                      .map(JSONValue::asString)
+                                      .map(Hash::new)
+                                      .collect(Collectors.toList());
+        var author = new Author(o.get("author_name").asString(),
+                                o.get("author_email").asString());
+        var authored = ZonedDateTime.parse(o.get("authored_date").asString());
+        var committer = new Author(o.get("committer_name").asString(),
+                                   o.get("committer_email").asString());
+        var committed = ZonedDateTime.parse(o.get("committed_date").asString());
+        var message = Arrays.asList(o.get("message").asString().split("\n"));
+        return new CommitMetadata(hash, parents, author, authored, committer, committed, message);
+    }
+
+    @Override
+    public Optional<HostedCommitMetadata> search(Hash hash) {
+        var hex = hash.hex();
+        for (var group : groups) {
+            var projects = request.get("groups/" + group + "/projects")
+                                  .execute()
+                                  .stream()
+                                  .map(o -> o.get("id").asString())
+                                  .collect(Collectors.toList());
+            for (var project : projects) {
+                var c = request.get("projects/" + project + "/repository/commits/" + hex)
```

Comment:
Could we use this endpoint instead? https://docs.gitlab.com/ce/api/commits.html#get-a-single-commit

Reply:
That is the endpoint the code is using :smile_cat: Did you perhaps miss the `+ hex` part?

Reply:
Well then, carry on!